packages/devtools_app/lib/src/shared/globals.dart

```diff
@@ -15,6 +17,22 @@ import 'framework_controller.dart';
 import 'preferences.dart';
 import 'survey.dart';
 
+/// Whether this DevTools build is external.
+bool get isExternalBuild => _isExternalBuild;
+bool _isExternalBuild = false;
+
+/// Flag the build as external.
+set isExternalBuild(bool value) {
+  if (kReleaseMode && _isExternalBuild && !value) {
```

Comment:
we should actually never be changing the value of isExternalBuild, regardless of what run mode we are in. So can we change this back to:

`late final bool isExternalBuild;`

Reply:
It should be false for test. Should we add it to all tests as setup?

Reply:
why should this be false for tests? the only place this should be false is in google3, where we don't run tests.

Reply:
I agree that we want to make sure experimental features are off in tests unless we are intentionally testing them. But flipping `isExternalBuild` to false is a hacky way to do this that could have other side effects in the code. Our tests test the "external" build, so this will always be true outside of google3.

Reply:
Good point. Set it to true by default.